testsuite/features/step_definitions/common_steps.rb

```diff
@@ -75,31 +75,33 @@
-  uptime_hours, _code = node.run("awk '{print $1/3600}' /proc/uptime")
-  uptime_minutes, _code = node.run("awk '{print $1/60}' /proc/uptime")
-  uptime_seconds, _code = node.run("awk '{print $1}' /proc/uptime")
-  uptime_days = uptime_days.to_f.round
-  uptime_hours = uptime_hours.to_f.round
-  uptime_minutes = uptime_minutes.to_f.round
-  uptime_seconds = uptime_seconds.to_f.round
-
-  if uptime_days < 2 and uptime_days >= 1
-    step %(I should see a "#{uptime_days.round} day ago" text)
-  elsif uptime_days < 1 and uptime_hours >= 2
-    step %(I should see a "#{uptime_hours.round} hours ago" text)
-  elsif uptime_days < 1 and uptime_hours < 2
-    step %(I should see a "#{uptime_hours.round} hour ago" text)
-  elsif uptime_hours < 1 and uptime_minutes >= 2
-    step %(I should see a "#{uptime_minutes.round} minutes ago" text)
-  elsif uptime_hours < 1 and uptime_minutes < 2
-    step %(I should see a "#{uptime_minutes.round} minute ago" text)
-  elsif uptime_minutes < 1 and uptime_seconds >= 2
-    step %(I should see a "#{uptime_seconds.round} seconds ago" text)
-  elsif uptime_minutes < 1 and uptime_seconds < 2
-    step %(I should see a "#{uptime_seconds.round} second ago" text)
+  uptime_seconds, _return_code = node.run("awk '{print $1}' /proc/uptime") # run code on node only once, to get uptime in seconds
+  uptime_days = (uptime_seconds / 86400.0) # 60 seconds * 60 minutes * 24 hours; the .0 forces a float division
+  uptime_hours = (uptime_seconds / 3600.0) # 60 seconds * 60 minutes
+  uptime_minutes = (uptime_seconds / 60.0) # 60 seconds
+
+  # rounded values to nearest integer number
+  rounded_uptime_minutes = uptime_minutes.round
+  rounded_uptime_hours = uptime_hours.round
+  rounded_uptime_days = uptime_days.round
+
+  # the moment.js library being used has some weird rules, which these conditionals follow
+  if (uptime_days >= 1 && rounded_uptime_days < 2) || (uptime_days < 1 && rounded_uptime_hours >= 22) # shows "a day ago" after 22 hours and before it's been 1.5 days
+    step %(I should see a "a day ago" text)
+  elsif rounded_uptime_hours > 1 && rounded_uptime_hours <= 21
+    step %(I should see a "#{rounded_uptime_hours} hours ago" text)
+  elsif rounded_uptime_minutes >= 45 && rounded_uptime_hours == 1 # shows "an hour ago" from 45 minutes onwards up to 1,5 hours
+    step %(I should see a "an hour ago" text)
+  elsif rounded_uptime_minutes > 1 && rounded_uptime_hours < 1
+    step %(I should see a "#{rounded_uptime_minutes} minutes ago" text)
+  elsif uptime_seconds >= 45 && rounded_uptime_minutes == 1
+    step %(I should see a "a minute ago" text)
+  elsif uptime_seconds < 45
+    step %(I should see a "a few seconds ago" text)
+  elsif rounded_uptime_days < 25
+    step %(I should see a "#{rounded_uptime_days} days ago" text) # shows "a month ago" from 25 days onwards
   else
-    step %(I should see a "#{uptime_days.round} days ago" text)
+    step %(I should see a "a month ago" text)
```

Comment:
LGTM.
I would have a variable for the text to search and then only call once the `step %(I should see a "#{my_variable}" text)` but is a matter of taste :)